### PR TITLE
[IMP] sale: simplify API of create_product_variant method

### PR DESCRIPTION
**Pull request by amh-mw**
### Current behavior before PR:

The `product.template` model has a nice function `create_product_variant` that unfortunately expects its input to be in JSON.

### Desired behavior after PR is merged:

JSON decoding is moved into the controller, such that the function can be called as expected.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sale/controllers/variant.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
-# -*- coding: utf-8 -*-
```
Comment (Feyensv) [inline addons/sale/controllers/variant.py:1]: Unrelated diff, leave the line ;)

**Review by Feyensv — COMMENTED** [review]
Ok for me, two little remarks and it's good to go :)

File: addons/sale/controllers/variant.py (hunk 1 above)
Comment (amh-mw) [reply addons/sale/controllers/variant.py:1]: Sure thing. It's a Python2-ism, thought I would get it since I was touching the file.
